src/libPersistence/BlockStorage.cpp

```diff
@@ -491,6 +570,8 @@ bool BlockStorage::GetAllTxBodiesTmp(std::list<TxnHash>& txnHashes) {
 
   LOG_MARKER();
 
+  shared_lock<shared_timed_mutex> g(m_mutexTxBodyTmp);
+
   leveldb::Iterator* it =
       m_txBodyTmpDB->GetDB()->NewIterator(leveldb::ReadOptions());
```

Comment:
Maybe let `m_mutexTxBodyTmp` to protect `leveldb::Iterator* it = m_txBodyTmpDB->GetDB()->NewIterator(leveldb::ReadOptions());` only

Reply:
cannot do that...the iterator is keep changing in the for loop